.github/workflows/version-bump.yml

```diff
@@ -0,0 +1,117 @@
+      - name: use node.js 16.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16.x
+          registry-url: https://registry.npmjs.org/ # Needed for auth
+      - name: cache all node_modules
+        id: cache-modules
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('yarn.lock', '**/package.json') }}
+      - name: find location of global yarn cache
+        id: yarn-cache
+        if: steps.cache-modules.outputs.cache-hit != 'true'
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - name: cache global yarn cache
+        uses: actions/cache@v2
+        if: steps.cache-modules.outputs.cache-hit != 'true'
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: yarn install
+        run: yarn install --frozen-lockfile
+      # End of yarn setup
+
+      - name: 'Set release name'
+        id: set_release_name
+        run: node scripts/set-release-name.js
+      - name: 'Configure git'
+        run: |
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "github-actions[bot]"
+      - name: 'Create topic branch'
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+          BRANCH=$(git branch --list topic/v${{ steps.set_release_name.outputs.release_version }})
+          if [ ! -z "$BRANCH" ]; then
+              git branch -D topic/v${{ steps.set_release_name.outputs.release_version }}
+          fi
+          git checkout -b topic/v${{ steps.set_release_name.outputs.release_version }}
+      - name: Run backstage-cli versions:bump --release ${{ inputs.release_line || 'next' }} command
+        run: yarn backstage-cli versions:bump --release ${{ inputs.release_line || 'next' }}
+      - name: 'Check for changes'
+        id: check_for_changes
+        run: |
+          CHANGES=$(git status --porcelain)
+          if [ -z "$CHANGES" ]; then
+              echo "No changes"
+              echo "HAS_CHANGES=0" >> $GITHUB_OUTPUT
+          else
+              echo "Has changes"
+              echo "HAS_CHANGES=1" >> $GITHUB_OUTPUT
+          fi
+      - name: 'Commit changes'
+        if: ${{ steps.check_for_changes.outputs.HAS_CHANGES == 1 }}
+        run: |
+          git status
+          git add -A -- :!.npmrc
+          git commit -m "v${{ steps.set_release_name.outputs.release_version }} version bump"
+          git push origin topic/v${{ steps.set_release_name.outputs.release_version }}
+      - name: 'Create Pull Request'
+        if: ${{ steps.check_for_changes.outputs.HAS_CHANGES == 1 }}
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{secrets.VERSION_BUMP_TOKEN}}
```

Comment:
I wonder if this is needed given that you can configure the default access token to have write access. Have you tried this out in your fork having write access configured for the token? https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-the-default-github_token-permissions
<img width="880" alt="Screenshot 2022-11-14 at 08 51 07" src="https://user-images.githubusercontent.com/68980/201604725-c650ccd5-5456-4200-b12a-0685f9e9a7a9.png">

Reply:
So the `GITHUB_TOKEN` will create the PR the problem is that won't trigger any of the related actions so the `pull.yml` won't fire which is nice 2nd layered of validation. Here's the docs on this little conundrum: https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow

Reply:
As discussed I've updated this to use the GH_SERVICE_ACCOUNT_TOKEN token